### PR TITLE
feat: super-components workspace and msc-tile super-component

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-11577

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11577 / MANAGER-11578 / MANAGER-11436
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -38,13 +38,14 @@ module.exports = {
...
       rules: {
+        'no-underscore-dangle': 'off',
         '@typescript-eslint/explicit-module-boundary-types': 'off',
         '@typescript-eslint/no-empty-function': 'off',
         'import/extensions': 'off',
         'no-use-before-define': 'off',
         'no-shadow': 'off',
         'no-unused-expressions': 'off',
-        'camelcase': 'off',
+        camelcase: 'off',
```
Comment (aboungnaseng-ovhcloud) [inline .eslintrc.js:48]: ?

Comment (chipp972) [reply .eslintrc.js:48]: autofix on javascript files

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

File: packages/manager/modules/veeam-enterprise/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
   "peerDependencies": {
     "@ovh-ux/manager-core": "^12.0.0 || ^13.0.0",
     "@ovh-ux/manager-ng-layout-helpers": "^2.0.0",
+    "@ovh-ux/ng-translate-async-loader": "^2.1.6",
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/modules/veeam-enterprise/package.json:23]: it's an eslint error

File: packages/manager/modules/pci/package.json [hunk 3]
```diff
@@ -43,6 +43,7 @@
     "@ovh-ux/manager-telecom-styles": "^4.0.0",
     "@ovh-ux/manager-trusted-nic": "^0.0.0 || ^1.0.0",
     "@ovh-ux/ng-at-internet": "^5.9.1",
+    "@ovh-ux/ng-at-internet-ui-router-plugin": "^3.2.3",
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/modules/pci/package.json:46]: it's an eslint error also

File: packages/super-components/components/msc-tile/react/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+dist/
+src/
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/react/.gitignore:2]: **question:** Why `src` folder is included in `.gitignore`?

File: packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    .tile-img {
+      height: auto;
+      max-width: 100%;
+      margin: 10px auto;
+    }
+
+    .tile-type {
+      display: block;
+      margin-bottom: 16px;
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss:23]: **nitpick:** To maintain uniformity across the Manager, use utility classes from Bootstrap

File: packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    .tile-badge-list {
+      display: inline-flex;
+      margin-left: 10px;
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss:28]: **nitpick:** To maintain uniformity across the Manager, use utility classes from Bootstrap

File: packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    .tile-title {
+      display: block;
+      margin-bottom: 16px;
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss:33]: **nitpick:** To maintain uniformity across the Manager, use utility classes from Bootstrap

File: packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    .tile-description {
+      display: block;
+      margin-bottom: 12px;
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss:38]: **nitpick:** To maintain uniformity across the Manager, use utility classes from Bootstrap

File: packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    .link-icon {
+      margin-left: 10px;
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/src/msc-tile/msc-tile.scss:42]: **nitpick:** To maintain uniformity across the Manager, use utility classes from Bootstrap

File: packages/super-components/components/msc-tile/vue/.gitignore [hunk 10]
```diff
@@ -0,0 +1,2 @@
+dist/
+src/
```
Comment (anooparveti) [inline packages/super-components/components/msc-tile/vue/.gitignore:2]: **question:** Why `src` folder is added to `.gitignore`?

**Review by anooparveti — COMMENTED** [review]
**suggestion:** Maybe we need to add organisation name to the prefix of supercomponent.
`msc` -> `omsc`

File: packages/super-components/generator/component/index.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        validate: (name) => name.length > 1,
+      },
+      {
+        type: 'input',
+        name: 'componentTag',
+        message: 'What is the tag of the new component ?',
+        default: 'msc-hello',
+        validate: (componentTag) =>
+          componentTag.length > 1 &&
+          !currentComponentList.includes(componentTag),
```
Comment (anooparveti) [inline packages/super-components/generator/component/index.js:49]: **question:** Are we restricting all the super-component tag to start with `msc-`. 
If yes, we can add this check to validate function.
If not, we should consider restricting.

Comment (anooparveti) [inline packages/super-components/generator/component/index.js:49]: **question:** Are we restricting all the super-component tag to start with `msc-`. 
If yes, we can add this check to validate function.
If not, we should consider restricting.

File: packages/super-components/generator/component/index.js [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+      },
+      {
+        type: 'input',
+        name: 'packageName',
+        message:
+          'What is the package name of the new component (without msc-) ?',
+        validate: (packageName) => packageName.length > 1,
+        default: ({ componentTag }) => componentTag.split('-')[1],
```
Comment (anooparveti) [inline packages/super-components/generator/component/index.js:57]: **nitpick:** In the `componentTag`, I dont see any restrictions that the tag must contain `-`. So this check can fail if the `componentTag` doesn't contain `-`

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/super-components/generator/component/templates/src/index.html.hbs [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html dir="ltr" lang="en">
+    <head>
+        <meta charset="utf-8" />
+        <meta
+            name="viewport"
+            content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
+        />
+        <title>Dev msc-tile</title>
```
Comment (anooparveti) [inline packages/super-components/generator/component/templates/src/index.html.hbs:9]:
```suggestion
        <title>Dev msc-{{componentTag}}</title>
```

File: packages/super-components/storybook/stories/Introduction.mdx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+#### Structure
+
+A super-component contains different files and folders :
+
+- `docs/` contains the stories that will be displayed in the storybook
+- `react/` the generated react component (you should not touch anything in this folder)
+- `screenshot/` the generated screenshots of the e2e tests
+- `src/` the source code of the component
+- `tests/` the folder containing the tests
+- `vue/` the generated react component (you should not touch anything in this folder)
```
Comment (anooparveti) [inline packages/super-components/storybook/stories/Introduction.mdx:81]:
```suggestion
- `vue/` the generated vue component (you should not touch anything in this folder)
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/super-components/components/msc-tile/src/index.html [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+                data-tracking="home:dashboard:product"
+            >
+                <span slot="badges">
+                    <osds-chip color="primary" size="sm">OVHcloud</osds-chip>
+                    <osds-chip color="success" size="sm">Beta</osds-chip>
+                </span>
+                <div slot="footer">
+                    <osds-button
+                        color="primary"
+                        style="margin-top: 1.5rem"
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/super-components/components/msc-tile/src/index.html:46]: the index.html file inside mcs-tile is only an exemple how the implement the msc-tile component,  we can use this component like that or sure we can create a css file to add the rule

File: packages/super-components/components/msc-tile/docs/create-tile.ts [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+          <osds-chip color="success" size="sm">Beta</osds-chip>
+        </span>
+      `
+          : ''
+      }
+      ${
+        hasFooter
+          ? `
+        <div slot="footer">
+          <osds-button color="primary" style="margin-top: 1.5rem">Commander</osds-button>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/super-components/components/msc-tile/docs/create-tile.ts:54]: same comment as below its only a css for documentation, of course we can create a file for this, but this way is more ligther to load the documentation  and maintain it

File: packages/super-components/generator/component/templates/vue/.gitignore [hunk 17]
```diff
@@ -0,0 +1,2 @@
+dist/
+src/
```
Comment (anooparveti) [inline packages/super-components/generator/component/templates/vue/.gitignore:2]: **nitpick:** Can you remove this from `.gitignore`?

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

**Review by Kranysys — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**